proposals/runtime_on_invalidated.md

```diff
@@ -0,0 +1,156 @@
+# Proposal: browser.runtime.onInvalidated event>
+
+** How to Use This Template **
+
+See [Proposal Process](proposal_process.md) for the detailed process on how to
+propose new APIs and use this template.  Each section includes instructions on
+what to include.  Delete the instructions for a given section once it's filled
+out.  Remove this section once the template is filled out.
+
+**Summary**
+
+`browser.runtime.onInvalidated`
+
+This event allows extension frames and contentScripts to be notified when
+background scripts get invalidated.
+
+**Document Metadata**
+
+**Author:** carlosjeurissen
+
+**Sponsoring Browser:** Chrome
+
+**Contributors:** TBD
+
+**Created:** 2025-03-26
+
+**Related Issues:** https://github.com/w3c/webextensions/issues/138
+
+## Motivation
+
+### Objective
+
+This event allows extension frames and contentScripts to be notified when
+background scripts get invalidated.
+
+#### Use Cases
+
+Cleanup of effects in contentScripts. After-invalidated handling of frames.
+For example, know when to use browser.tabs vs window.open.
+
+### Known Consumers
+
+Any extension with a content script which applies certain side effects to the
+web page.
+
+## Specification
+
+### Schema
+
+New onInvalidated event:
+```json
+{
+  "name": "onInvalidated",
+  "type": "function",
+  "description": "Fired when the extension background scripts get invalidated.",
+  "parameters": [
+    {
+      "type": "object",
+      "name": "details",
+      "properties": {
+        "reason": {
+          "optional": true,
```

Comment:
Why is this optional?

Reply:
Did anyone express that they would do this? I'd rather have a consistent set of properties unless there's a compelling reason otherwise.

Reply:
@oliverdunk I do not believe any browser vendor stated any kind of in favour / opposed for the reason property. So this is left purposely open. Talking of which, would Chrome be in favour / opposed for the reason property?

Reply:
We're supportive. As discussed, we'd prefer to remove the `optional` modifier here.

Reply:
@oliverdunk updated to not be optional.